src/components/views/SelectContact.tsx

```diff
@@ -0,0 +1,123 @@
+import ContactTableRowItem from '../modals/ContactsModal/ContactTableRowItem';
+import useContacts from '@/hooks/boardwalk/useContacts';
+import { UserPlusIcon } from '@heroicons/react/20/solid';
+import { Button, Table, TextInput } from 'flowbite-react';
+import { useMemo, useState } from 'react';
+import { PublicContact } from '@/types';
+import { useToast } from '@/hooks/util/useToast';
+import useGifts from '@/hooks/boardwalk/useGifts';
+
+const SelectContact = ({
+   onSelectContact,
+}: {
+   onSelectContact: (contact: PublicContact) => void;
+}) => {
+   const [currentView, setCurrentView] = useState<'select' | 'add'>('select');
+   const [addingContact, setAddingContact] = useState(false);
+   const [searchTerm, setSearchTerm] = useState('');
+   const [addContactInput, setAddContactInput] = useState('');
+   const { sortedContacts, user, addContact, fetchContactByUsername, isContactAdded } =
+      useContacts();
+   const { addToast } = useToast();
+   const { loadUserCustomGifts } = useGifts();
+
+   const filteredContacts = useMemo(() => {
+      const f = [...sortedContacts, user].filter(contact =>
```

Comment:
it's probably fine. I just wasn't' sure what it is because I guess I don't understand why would you show yourself under contacts

Reply:
In case you want to lock ecash to yourself. I'm probably the only one that uses that, but its kinda fun to send someone locked ecash they can't spend just to show them.

Reply:
I see. Makes sense to keep it then